### PR TITLE
fix Query arguments with ^ not working

### LINKED ISSUE #2070: Query arguments with ^ not working
``` 
$ rpm -qa|grep '\^'
python3-jedi-0.17.2^20200805git209e271-2.fc34.noarch
pythran-0.9.8^post3-3.fc34.x86_64
$ rpm -q 'python3-jedi-0.17.2^20200805git209e271-2.fc34.noarch'
package python3-jedi-0.17.2^20200805git209e271-2.fc34.noarch is not installed
$ rpm -q 'pythran-0.9.8^post3-3.fc34.x86_64'                   
package pythran-0.9.8^post3-3.fc34.x86_64 is not installed
```

It seems like rpm does one layer of unexpected unescaping somewhere:
```
$ rpm -q 'python3-jedi-0.17.2\^20200805git209e271-2.fc34.noarch'
python3-jedi-0.17.2^20200805git209e271-2.fc34.noarch
$ rpm -q 'pythran-0.9.8\^post3-3.fc34.x86_64'
pythran-0.9.8^post3-3.fc34.x86_64
```

(Note that '^' is not special for the shell, so '\^' gets passed as two characters to rpm, a backslash and the caret.)

Originally reported at https://bugzilla.redhat.com/show_bug.cgi?id=2002038

### PR DESCRIPTION
**Pull request by yangchenguang94**
fix  #2070 
Use the same method as +

### CODE REVIEW

**Comment by yangchenguang94** [comment]
image

**Comment by pmatilai** [comment]
This seems closely related to #2069, @dmnks gets to be our appointed escape-specialist from now on :grin:

**Comment by yangchenguang94** [comment]
> This seems closely related to #2069, @dmnks gets to be our appointed escape-specialist from now on 😁

I am a tester. This is the first time  use gdb to debug code. hope my results will be recognized and encourage me to make continuous contributions to the community 😁

**Comment by ffesti** [comment]
Tested. Fixes the issue and is the obvious right thing to do.

I wrote a test case for this that I will submit in a separate PR.

File: lib/rpmdb.c [hunk 1]
```diff
@@ -1133,6 +1133,7 @@ static char * mireDup(rpmTagVal tag, rpmMireMode *modep,
 	    switch (*s) {
 	    case '.':
 	    case '+':
+            case '^':
```
[the remark this answers is not in the document]
Comment (yangchenguang94) [reply lib/rpmdb.c]: I got it

**Comment by ffesti** [comment]
Thanks a lot for your work and quick response! Great work for someone just getting started with this!

**Comment by ffesti** [comment]
See #2123 for the test cases - if anyone is interested

**Comment by soig** [comment]
It's not fixed in the rpm-4.18.x branch AFAIK, only in master.

**Comment by AdamWill** [comment]
very minor note: when updating these blocks, the comments haven't been updated. This comment:

    /* periods and plusses are escaped, splats become '.*' */

could read:

    /* periods, plusses and carets are escaped, splats become '.*' */

and this one:

    /* Copy pattern, escaping periods, prefixing splats with period. */

could read:

    /* Copy pattern, escaping periods, plusses and carets, and prefixing splats with period. */

**Pull request merged**